frontend/public/components/utils/nav-title.tsx

```diff
@@ -36,9 +36,12 @@ export const NavTitle = connectToModel((props: NavTitleProps) => {
   const {kind, kindObj, detail, title, menuActions, buttonActions, obj, breadcrumbsFor, style} = props;
   const data = _.get(obj, 'data');
   const isCSV = !_.isEmpty(data) && referenceFor(data) === referenceForModel(ClusterServiceVersionModel);
+  const isCSC = !_.isEmpty(data) && referenceFor(data) === referenceForModel(ClusterServiceClassModel);
+  const isCSP = !_.isEmpty(data) && referenceFor(data) === referenceForModel(ClusterServicePlanModel);
+  const displayName = !isCSC && !isCSP ? title : (isCSC ? serviceClassDisplayName(data) : servicePlanDisplayName(data));
```

Comment:
nav-title should have no knowledge of the service catalog resources. We need to find a better way that is not so tightly coupled. It would be better to have some kind of optional callback function to get the title

Reply:
> Looks like you could pass an optional displayName callback function to DetailsPage then on to NavTitle

Ok, implemented, -thanks